internal-functions

```diff
@@ -75,3 +75,49 @@ fn-acl-check-app() {
     dokku_log_fail "You can only modify ACL using local dokku command on target host"
   fi
 }
+
+fn-acl-check-service() {
+  declare SERVICE_TYPE="$1" SERVICE="$2"
+
+  local SERVICE_PATH="$DOKKU_LIB_ROOT/services/$SERVICE_TYPE/$SERVICE"
+  if ! [[ -d $SERVICE_PATH ]]; then
+    dokku_log_fail "Service $SERVICE of type $SERVICE_TYPE does not exist"
+  fi
+
+  if [[ -n "${NAME:-}" ]]; then
+    dokku_log_fail "You can only modify ACL using local dokku command on target host"
+  fi
+}
+
+check-app-acl() {
+  declare desc="Checks if the current user has an ACL entry for the app"
+  declare APP=$1
+
+  if ! ( verify_app_name "$APP" 2>/dev/null ); then
+    dokku_log_fail "User $SSH_NAME does not have permissions to run $CMD on $APP, or $APP does not exist"
+  fi
+
+  ACL_FILE="$DOKKU_ROOT/$APP/acl/$SSH_NAME"
```

Comment:
The `SSH_NAME` should really be passed as an argument. That the plugin doesn't always do that would to me mean it needs cleanup, but at least for new code we should be explicit over implicit. Makes it easier to test by injecting arguments.